subsys/settings/src/settings_line.c

```diff
@@ -470,7 +470,7 @@ static int settings_line_cmp(char const *val, size_t val_len,
 	size_t len_read, exp_len;
 	size_t rem;
 	char buf[16];
-	int rc;
+	int rc = 0;
```

Comment:
same as above

Reply:
I also noted that return value could be 1 if can not to finnd until_char, and this is not described in API documentation.